node/bft/src/bft.rs

```diff
@@ -79,6 +80,7 @@ pub struct BFT<N: Network> {
 impl<N: Network> BFT<N> {
     /// Initializes a new instance of the BFT.
     pub fn new(
+        block_sync: Arc<BlockSync<N>>,
```

Comment:
while I'm in favor of keeping the `BlockSync` in a single `Arc`, this should be followed by the removal of the individual `Arc`s in its members, and its `Clone` derivation; otherwise this will reduce the number of atomic ops related to its cloning, but it could suggest that `BlockSync` itself or its members don't derive `Clone`

Reply:
I actually didn't know about the `derive(Clone)` in many places. Do we prefer `Arc` or making structs clonable? Or is both okay?

Reply:
1. if we need the members to be have shared ownership individually, they need their own `Arc` wrappers
2. it is more efficient to have one such wrapper over an object that doesn't derive `Clone` and whose members don't have `Arc`s (if they don't need to individually)
3. having multiple "levels" of `Arc` isn't inherently inefficient, but it may be counterintuitive
4. if there's an object that derives `Clone` and has multiple `Arc`ed members, cloning it involves several atomic operations

Apologies if this is a bit convoluted; my preference is for fewer atomic operations on cloning (fewer `Arc`s involved in a single clone op), and clarity (i.e. not `Arc`ing someone that is never handled individually).

Reply:
Thanks! I have the same preference, but saw code that has multiple arced members and derives clone, so I wasn't sure which is the desired code style.